### ISSUE TITLE
flatpak-builder: --device can be more specific

### ISSUE THREAD

**Issue by marcosps** [flatpak/flatpak#982]
When creating a new json manifest for my webcam app, I needed to add "--device=all" (other apps like Steam also do that).

In this case, should flatpak have more targets from --device, like --device=media (for webcams on /dev/videoX and radio devices for /dev/radioX) ?

Currently, only kvm, dri and all exists (https://github.com/flatpak/flatpak/blob/2a661fd448a153992e07904b666ef4fb748810a3/common/flatpak-run.c#L112)

If you guys agree, I can work on that.

**Comment by alexlarsson** [comment]
For webcams the plan is to use pipewire: https://github.com/flatpak/flatpak/issues/83
We want to avoid static permissions as much as possible, and only have it as a fallback when nothing else is possible.

**Comment by marcosps** [comment]
Do you have any code pointers to start looking at this? There are some people interested in this feature, by looking at the issue you pointed, but I didn't found any code about it yet.

**Comment by alexlarsson** [comment]
Pipewire itself is here: https://github.com/wtay/pipewire
But i have not looked at integration with it yet, so there is no code for that. Its on the todo list for 1.0 though.

**Comment by marcosps** [comment]
Yes, I looked at this repository yesterday, and already sent a pull request, but couldn't execute any example... maybe more documentation about how to run a simple example would be good for new comers. So, more work to do :)

**Comment by amtlib-dot-dll** [comment]
Also a `--device=usb` to `--dev-bind` device files like `/dev/bus/usb` and more, to allow specifically USB access for devtools like Arduino or STM32

**Comment by TingPing** [comment]
I think `joystick` would be a reasonable addition as it shouldn't really be a security problem, nothing exists as a portal for it yet, and is widely using `all` currently which is pretty bad.
